src/App.tsx

```diff
@@ -15,48 +16,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  alphabetically = 'alphabet',
+  byLength = 'length',
+  default = '',
+}
+
+type SortOptions = {
+  sortField: SortType,
+  reverseField: boolean,
+};
+
+function getSortedGoods(
+  goods: string[],
+  { sortField, reverseField }: SortOptions,
+) {
+  const sortedGoods = [...goods];
+
+  if (sortField) {
+    sortedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.alphabetically:
+          return good1.localeCompare(good2);
+        case SortType.byLength:
+          return good1[SortType.byLength]
+            - good2[SortType.byLength];
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverseField) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState(SortType.default);
+  const [reverseField, setReverseField] = useState(false);
+  const visibleGoods = getSortedGoods(
+    goodsFromServer,
+    { sortField, reverseField },
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SortType.alphabetically,
+          })}
+          onClick={() => setSortField(SortType.alphabetically)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button', 'is-success', {
+            'is-light': sortField !== SortType.byLength,
+          })}
+          onClick={() => setSortField(SortType.byLength)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn('button', 'is-warning', {
+            'is-light': reverseField === false,
+          })}
+          onClick={() => setReverseField(!reverseField)}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortField || reverseField) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortField(SortType.default);
+              setReverseField(false);
+            }}
```

Comment:
it is bad practise to make anonymous arrow functions in jsx.
make handler in component and pass reference here